BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.h

```diff
@@ -0,0 +1,153 @@
+ * @param sessionTaskId         Id of the session task
+ * @param resumeData            resume data to resume background download session task from
+ * @param error                 error if fail to cache
+ *
+ * @return YES if succeeded, NO if failed
+ */
+- (BOOL)cacheBackgroundSessionId:(NSString *)backgroundSessionId sessionTaskId:(NSUInteger)sessionTaskId resumeData:(NSData *)resumeData error:(NSError **)error;
+
+/**
+ * Cache responseData of a background session task
+ *
+ * @param backgroundSessionId   Id of the background session
+ * @param sessionTaskId         Id of the session task
+ * @param responseData          response data from the session task
+ * @param error                 error if fail to cache
+ *
+ * @return YES if succeeded, NO if failed
+ */
+- (BOOL)cacheBackgroundSessionId:(NSString *)backgroundSessionId sessionTaskId:(NSUInteger)sessionTaskId responseData:(NSData *)responseData error:(NSError **)error;
+
+/**
+ * Cache response of a background session task
+ *
+ * @param backgroundSessionId   Id of the background session
+ * @param sessionTaskId         Id of the session task
+ * @param response              NSURLResponse from the session task
+ * @param error                 error if fail to cache
+ *
+ * @return YES if succeeded, NO if failed
+ */
+- (BOOL)cacheBackgroundSessionId:(NSString *)backgroundSessionId sessionTaskId:(NSUInteger)sessionTaskId response:(NSURLResponse *)response error:(NSError **)error;
+
+/**
+ * Cache client-side error of a background session task
+ *
+ * @param backgroundSessionId   Id of the background session
+ * @param sessionTaskId         Id of the session task
+ * @param taskError             store client-side NSError of the session task
+ * @param error                 error if fail to cache
+ *
+ * @return YES if succeeded, NO if failed
+ */
+- (BOOL)cacheBackgroundSessionId:(NSString *)backgroundSessionId sessionTaskId:(NSUInteger)sessionTaskId taskError:(NSData *)taskError error:(NSError **)error;
+
+/**
+ * Get all cached data of the background session task associated with this userId and associateId
+ *
+ * @param userId            Id of user started the session task
+ * @param associateId       Id to associate with the session task
+ * @param error             error if fail to get
+ *
+ * @return BOXURLSessionTaskCache   all cached data of the session task
+ */
+- (BOXURLSessionTaskCachedInfo *)getCacheForUserId:(NSString *)userId associateId:(NSString *)associateId error:(NSError **)error;
+
+/**
+ * Delete all cached data of the background session task associated with this userId and associateId
+ *
+ * @param userId            Id of user started the session task
+ * @param associateId       Id to associate with the session task
+ * @param error             error if fail to delete
+ *
+ * @return YES if succeeded, NO if failed
+ */
+- (BOOL)deleteCachedInfoForUserId:(NSString *)userId associateId:(NSString *)associateId error:(NSError **)error;
```

Comment:
Minor: Apple's naming convention involves capitalizing full acronyms (like URL), in this case ID. Also, for ID Apple tends to go with 'Identifier'